backend/layers/processing/process_download.py

```diff
@@ -187,5 +187,6 @@ def process(
         )
         self.update_processing_status(dataset_version_id, DatasetStatusKey.UPLOAD, DatasetUploadStatus.UPLOADED)
 
-        sfn_client = StepFunctionProvider().client
-        sfn_client.send_task_success(taskToken=sfn_task_token, output=json.dumps(response))
+        if sfn_task_token:
```

Comment:
for unit test? The only thing is this could cause a silent failure in deployed scenario. Patch instead?

Reply:
this job isn't part of a step function, so there's nothing to send task success to nor a need to init a StepFunctionClient. But I wanted to re-use the rest of the `process_download.process` logic

I can extract everything above this line into a separate function and call that instead, though

Reply:
Ah, right. I would just add a comment explaining that for lesser minds like mine 😅